bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/ZkLedgerUnderreplicationManager.java

```diff
@@ -884,4 +886,51 @@ public String getReplicationWorkerIdRereplicatingLedger(long ledgerId)
         }
         return replicationWorkerId;
     }
+
+    @Override
+    public void setCheckAllLedgersCTime(long checkAllLedgersCTime) throws UnavailableException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("setCheckAllLedgersCTime");
+        }
+        try {
+            CheckAllLedgersFormat.Builder builder = CheckAllLedgersFormat.newBuilder();
+            builder.setCheckAllLedgersCTime(checkAllLedgersCTime);
+            if (zkc.exists(checkAllLedgersCtimeZnode, false) != null) {
+                zkc.setData(checkAllLedgersCtimeZnode, TextFormat.printToString(builder.build()).getBytes(UTF_8), -1);
```

Comment:
What about using binary format? This is new stuff, better not to start with forward/backward compatibility problems

Reply:
changing to binary format